content/documentation/architecture.adoc

```diff
@@ -0,0 +1,116 @@
+---
+title: "Architecture"
+date: 2018-09-03T14:39:48-05:00
+draft: false
+---
+
+= Kiali architecture
+:imagesdir: /images/documentation/architecture
+
+Kiali is composed of two components: a backend running in the container
+application platform, and a user frontend. However, in order to be useful,
+Kiali depends on external services and components provided by other
+applications.
+
+The following diagram illustrates the components involved in Kiali and its
+interactions:
+
+image::architecture.png[Kiali architecture]
+
+== Kiali backend
+
+Tha backend is the application that runs in the container application platform.
+It's written in http://golang.org/[Go]. The code can be found at
+https://github.com/kiali/kiali[kiali/kiali GitHub repository].
+
+This is the component that communicates with Istio parts, retrieves and
+processes data, and send processed data to the frontend.
+
+The backend doesn't need storage. When running the backend locally, all
+configurations are set in files and environment variables (including the user
+and password used for authentication). When deploying the backend to a cluster
+using the docker images, configmaps and secrets are used to set the
+configurations.
+
+== Kiali frontend
+
+The frontend is the application that runs in the browser.  It's a React
+application, written in Typescript. The code can be found at
+https://github.com/kiali/kiali-ui[kiali/kiali-ui GitHub repository].
+
+In a standard deployment, the backend provides the frontend, but once the
```

Comment:
Suggestion: since I think most people know or at least have heard about "single page app" , maybe we can avoid this explanation and just say it 's a single page app?

Reply:
I reworded this and also the previous paragraph to fix both this comment and @abonas comment. Please, check again to see if it's better.